roll/__init__.py

```diff
@@ -268,22 +270,25 @@ def __init__(self):
     def factory(self):
         return self.Protocol(self)
 
-    def route(self, path: str, methods: list=None):
+    def route(self, path: str, methods: list=None, **extras):
         if methods is None:
             methods = ['GET']
 
         def wrapper(func):
-            self.routes.add(path, **{m: func for m in methods})
+            handlers = {method: func for method in methods}
```

Comment:
That way we are mixing methods and extra in the same space.
Maybe what we should do: add a `has_route` method to `autoroute`, so we can make the payload merge our-selves, and thus have a `handlers` key.

Reply:
I agree that it may be the role of autoroutes to deal with that.